compiler/packages/babel-plugin-react-compiler/src/Entrypoint/Program.ts

```diff
@@ -238,23 +224,23 @@ function insertNewFunctionNode(
           loc: originalFn.node.loc ?? null,
         },
       );
+      const fn: t.FunctionDeclaration = {
+        type: 'FunctionDeclaration',
+        id: compiledFn.id,
+        loc: originalFn.node.loc ?? null,
+        async: compiledFn.async,
+        generator: compiledFn.generator,
+        params: compiledFn.params,
+        body: compiledFn.body,
+      };
       const varDecl = originalFn.parentPath.parentPath;
-      varDecl.insertAfter(
-        t.variableDeclaration('const', [
-          t.variableDeclarator(compiledFn.id, funcExpr),
-        ]),
-      );
-      const insertedFuncExpr = varDecl.get('declarations')[0]!.get('init')!;
-      CompilerError.invariant(
-        insertedFuncExpr.isArrowFunctionExpression() ||
-          insertedFuncExpr.isFunctionExpression(),
-        {
-          reason: 'Expected inserted (arrow) function expression',
-          description: `Got: ${insertedFuncExpr}`,
-          loc: insertedFuncExpr.node?.loc ?? null,
-        },
-      );
-      return insertedFuncExpr;
+      const insertedFuncDecl = varDecl.insertAfter(fn)[0]!;
```

Comment:
Since we're not using the `createNewFunctionNode` helper here, let's manually exclude from re-compilation (`ALREADY_COMPILED.add(transformedFn);`)